src/foam/parse/DateParser.java

```diff
@@ -0,0 +1,1079 @@
+          cal.set(Calendar.MILLISECOND, (Integer) result.get("millisecond"));
+        }
+        // Subtract offset
+        cal.add(Calendar.MINUTE, -offset);
+      } else {
+        // No timezone - use UTC as-is
+        cal.set(year, month, day, hour, minute, second);
+        if ( result.containsKey("millisecond") ) {
+          cal.set(Calendar.MILLISECOND, (Integer) result.get("millisecond"));
+        }
+      }
+
+      return cal.getTime();
+    } catch (RuntimeException e) {
+      throw e; // Re-throw RuntimeException as-is
+    } catch (Exception e) {
+      throw new RuntimeException("Unsupported DateTime format: " + str, e);
+    }
+  }
+
+  /**
+   * Convenience method: parseDateTimeUTC with default grammar
+   */
+  public Date parseDateTimeUTC(String str) {
+    return parseDateTimeUTC(str, null);
+  }
+
+  // ========== Helper Methods ==========
+
+  /**
+   * Flatten timezone array from parser into a string
+   */
+  private String flattenTimezone(Object tzArray) {
+    if ( tzArray == null ) return null;
+    if ( tzArray instanceof String ) return (String) tzArray;
+
+    // Handle array structure from parser
+    if ( tzArray instanceof Object[] ) {
+      StringBuilder result = new StringBuilder();
+      Object[] arr = (Object[]) tzArray;
+      for ( Object item : arr ) {
+        if ( item instanceof Object[] ) {
+          for ( Object subItem : (Object[]) item ) {
+            result.append(subItem);
+          }
+        } else if ( item != null ) {
+          result.append(item);
+        }
+      }
+      return result.toString();
+    }
+
+    return String.valueOf(tzArray);
+  }
+
+  /**
+   * Parse timezone string and return offset in minutes.
+   * Z means UTC (0). +05:30 means +330 minutes.
+   */
+  private int parseTimezone(String tz) {
+    if ( tz == null || tz.equals("Z") ) return 0;
+
+    int sign = tz.charAt(0) == '+' ? 1 : -1;
+    String nums = tz.substring(1).replace(":", "");
+
+    int hours, minutes;
+    if ( nums.length() >= 4 ) {
+      // HHMM format
+      hours = Integer.parseInt(nums.substring(0, 2));
+      minutes = Integer.parseInt(nums.substring(2, 4));
+    } else if ( nums.length() == 2 ) {
+      // HH format (no minutes)
+      hours = Integer.parseInt(nums);
+      minutes = 0;
+    } else {
+      return 0;
+    }
+
+    return sign * (hours * 60 + minutes);
+  }
+
+  /**
+   * Converts 2-digit year using fixed pivot at 50:
+   * - Years 00-49 map to 2000-2049
+   * - Years 50-99 map to 1950-1999
+   */
+  private int convertTwoDigitYear(int twoDigitYear) {
+    // Fixed pivot at 50
+    if ( twoDigitYear < 50 ) {
+      return 2000 + twoDigitYear;
+    }
+    return 1900 + twoDigitYear;
+  }
+
+  /**
+   * Converts 3-letter month abbreviation to 0-based month index (JAN→0, FEB→1, etc.)
+   */
+  private int parseMonthName(String monthName) {
+    String month = monthName.toUpperCase();
+    switch (month) {
+      case "JAN": return 0;
```

Comment:
If concerned about locale, this could be
return java.time.format.DateTimeFormatter.ofPattern("MMM")
      .withLocale(Locale.ENGLISH)
      .parse(monthName)
      .get(ChronoField.MONTH_OF_YEAR);